app/os_macos.go

```diff
@@ -514,7 +519,11 @@ func (w *window) EditorStateChanged(old, new editorState) {
 	}
 }
 
-func (w *window) ShowTextInput(show bool) {}
+func (w *window) ShowTextInput(show bool) {
+	if show && !w.config.Focused {
+		C.setFocus(w.view)
+	}
+}
```

Comment:
This is macOS, which doesn't have SoftwareKeyboard. However, I didn't find any specific function to request focus, and adding a new function will require changes on multiple OSes, even as `no-op`.

Reply:
I'm not sure why you need this here and not on other desktop platforms. Why can't `C.setFocus` be called automatically in the cases where focus was lost?

Reply:
It's possible to call it when the GioView is clicked, and doesn't have focus. I don't think Gio should immediate request for Focus, because it's possible to have multiple non-Gio View in the same Window.